apps/web/screens/UniversitySearch/UniversitySearch.tsx

```diff
@@ -0,0 +1,1252 @@
+      if (filters[str].length > 0) {
+        activeFiltersFound.push({ key, value: filters[str] })
+      }
+    })
+    //if no filters are active, then show no products
+    if (query === '' && activeFiltersFound.length === 0) {
+      resetFilteredList()
+    } else {
+      const results = SearchProducts({
+        fuseInstance,
+        query,
+        activeFilters: activeFiltersFound,
+      })
+
+      setFilteredResults(results)
+    }
+  }, [filters, query])
+
+  const resetFilteredList = () => {
+    const resultProducts: Array<Fuse.FuseResult<Program>> = data.map(
+      (item: Program, index: number) => {
+        return { item, refIndex: index, score: 1 }
+      },
+    )
+    setFilteredResults(resultProducts)
+  }
+
+  const createPrimaryCTA = (item: Program) => {
+    const now = new Date()
+    const CTA: CTAProps = {
+      label: n('apply', 'Sækja um'),
+      variant: 'primary',
+      size: 'small',
+      icon: 'arrowForward',
+      iconType: 'outline',
+      onClick: () => console.log('TODO'),
+      disabled:
+        new Date(item.applicationStartDate) > now ||
+        new Date(item.applicationEndDate) < now,
+    }
+    return CTA
+  }
+
+  const handleFilters = (filterKey: string, filterValue: string) => {
+    let str = filterKey as keyof typeof filters
+    if (filters[str].includes(filterValue)) {
+      const index = filters[str].indexOf(filterValue)
+      const specificArray = filters[str]
+      specificArray.splice(index, 1)
+      setFilters({ ...filters, [filterKey]: specificArray })
+    } else {
+      setFilters({
+        ...filters,
+        [filterKey]: [...filters[str], filterValue],
+      })
+    }
+  }
+
+  const handleComparisonChange = (dataItem: ComparisonProps) => {
+    let found = false
+    selectedComparison.forEach((x) => {
+      if (x.id === dataItem.id) {
+        found = true
+      }
+    })
+
+    if (!found) {
+      if (selectedComparison.length === MAX_SELECTED_COMPARISON) {
+        //comparison can only include 3 items so display error message if trying to add the fourth
+        toast.error(
+          n(
+            'maxComparisonError',
+            `Aðeins er hægt að hafa ${MAX_SELECTED_COMPARISON} nám í samanburði`,
+          ),
+        )
+      } else {
+        setSelectedComparison([...selectedComparison, dataItem])
+      }
+    } else {
+      setSelectedComparison(
+        selectedComparison.filter((item) => {
+          if (item.id !== dataItem.id) {
+            return true
+          }
+          return false
+        }),
+      )
+    }
+  }
+
+  useEffect(() => {
+    //update localStorage
+    localStorage.setItem('comparison', JSON.stringify(selectedComparison))
+  }, [selectedComparison])
+
+  useEffect(() => {
+    //update localStorage
+    localStorage.setItem('viewChoice', gridView ? 'true' : 'false')
+  }, [gridView])
+
+  const predefinedFilterOpenings: Array<boolean> = []
+  for (var x = 0; x < NUMBER_OF_FILTERS; x++) {
```

Comment:
or just 

`NUMBER_OF_FILTERS.forEach((n) => predefinedFilterOpenings.push(true));`

Reply:
NUMBER_OF_FILTERS is just a const number, you can't foreach a number, riiight?? or am i misunderstanding haha

Reply:
hahahah most certainly not! 🤣 
but we can do .fill() 

ps. now im just tryin to nitpick